### DISCUSSION TITLE
Is it possible to generate openapi.json with "nullable" : "true" for non-required property?

### DISCUSSION THREAD

**Discussion by dennismeissel** [fastapi/fastapi#7882] [Questions] [answered]
### Description

Is it possible to generate **openapi.json** with `"nullable" : "true"` for non-required property?

### Additional context

I'm using NSwag to generate C# code from FastAPI openapi.json. I need to set "nullable" as "true" to make it work correctly.

I have a FastAPI Model like this:

```python
class Topic(BaseModel):
    first_id: str = None
```

I want to get a schema like this:

```json
"Topic": {
  "title": "Topic",
  "type": "object",
    "properties": {
      "first_id": {
        "title": "First_Id",
        "type": "string",
        "nullable" : "true"
      }
  }
}
```
But I haven't found a way to set the field "nullable". And if it's not provided, it is "false" by default.

**Comment by nkhitrov** [comment]
You can use [Field](https://pydantic-docs.helpmanual.io/usage/schema/) to set `nullable=True` by default. 

Example:

```python

from fastapi import FastAPI, Depends, Query, HTTPException
from typing import Optional
from pydantic import BaseModel, Field

app = FastAPI()


class Model(BaseModel):
    a: Optional[int]
    b: Optional[int] = None
    c: Optional[int] = Field(None, nullable=True)


@app.get("/test", response_model=Model)
def foo(m: Model):
    return m
```

Component in `/openapi.json`
```
...
Model: {
    title: "Model",
    required: [
		"c"
	],
    type: "object",
    properties: {
    a: {
    	title: "A",
    	type: "integer"
	},
    b: {
    	title: "B",
    	type: "integer"
	},
    c: {
    	title: "C",
    	type: "integer",
    	nullable: true
	}
}
...
```

**Comment by dennismeissel** [comment, marked as answer]
@Slyfoxy Thanks for answer, it helped!

I found out, that `Field` [is not working with FastAPI](https://github.com/tiangolo/fastapi/issues/634#issuecomment-544564280)

But `Schema` did the job!

```python
from pydantic import BaseModel, Schema

class Topic(BaseModel):
    first_id: str = Schema(None, nullable=True)
```
And `/openapi.json`
```json
"Topic": {
    "title": "Topic",
    "type": "object",
    "properties": {
        "first_id": {
        "title": "First_Id",
        "type": "string",
        "nullable": true
        }
    }
}
```

**Comment by nkhitrov** [comment]
Fastapi works with 'Field' from 'pydantic' in [latest](https://fastapi.tiangolo.com/tutorial/body-fields/) version of 'fastapi' with 'pydantic=1.0'. Just update it :)

**Comment by dennismeissel** [comment]
Ah, ok, understood! 

Thank you :)

**Comment by tiangolo** [comment]
Thanks for the help here @Slyfoxy ! :cake: :bowing_man: 

Thanks @dennismeissel for reporting back and closing the issue :+1:
